njit_fastmath.py

```diff
@@ -0,0 +1,72 @@
+import pathlib
+
+from utils import check_callees, check_functions
+
+stumpy_path = pathlib.Path(__file__).parent / "stumpy"
+filepaths = sorted(f for f in pathlib.Path(stumpy_path).iterdir() if f.is_file())
+
+all_functions = {}
+
+ignore = ["__init__.py", "__pycache__"]
+for filepath in filepaths:
+    file_name = filepath.name
+    if file_name not in ignore and str(filepath).endswith(".py"):
+        prefix = file_name.replace(".py", "")
+
+        func_names, is_njit, fastmath_values = check_functions(filepath)
+        func_names = [f"{prefix}.{fn}" for fn in func_names]
+
+        all_functions[file_name] = {
+            "func_names": func_names,
+            "is_njit": is_njit,
+            "fastmath_values": fastmath_values,
+        }
+
+all_stumpy_functions = set()
+for file_name, file_functions_metadata in all_functions.items():
+    all_stumpy_functions.update(file_functions_metadata["func_names"])
+
+all_stumpy_functions = list(all_stumpy_functions)
```

Comment:
This feels strange. You declare `all_stumpy_functions` as a `set()` above and then you `update` it and then you convert it into a list all within a for-loop.

Reply:
One thing that I forgot to notice is that the nested dictionary `all_functions` has unique values for all_functions['file_name']['func_names"]. So, I can just use `list`.

```
all_stumpy_functions = []
for file_name in all_functions.keys():
    all_stumpy_functions.extend(all_functions['file_name']['func_names'])
```

Does that address your concern?

Reply:
It's not entirely clear what the goal is here

Reply:
Every time I look at it... I feel like the nested dictionary makes it hard to read. I need to re-design it.

> It's not entirely clear what the goal is here

My goal was to create a comprehensive list that contains all the functions that are defined in `./stumpy/`, and then use it to ignore any callee that is NOT in that comprehensive list. For instance, `range` can be a callee of a function. However, I do not want to keep it. I want to only have the callers/ callees that are defined in stumpy.